tests/unit/backend/corpora/api_server/curator/collection/test_dataset.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from backend.corpora.common.corpora_orm import CollectionVisibility, UploadStatus
+from tests.unit.backend.corpora.api_server.base_api_test import BaseAuthAPITest
+
+
+class TestDeleteDataset(BaseAuthAPITest):
+    def test__delete_dataset(self):
+        curator_tag = "tag.h5ad"
+        processing_status = {"upload_status": UploadStatus.UPLOADING, "upload_progress": 10.0}
+        queries = ["dataset_uuid", "curator_tag"]
+        auth_credentials = [
+            (self.make_super_curator_header, "super", 202),
+            (self.get_auth_headers, "owner", 202),
+            (None, "none", 401),
+            (self.make_not_owner_header, "not_owner", 403),
+        ]
+        for query in queries:
+            for auth, auth_description, expected_status_code in auth_credentials:
+                with self.subTest(f"{query} {auth_description} {expected_status_code}"):
+                    collection = self.generate_collection(self.session, visibility=CollectionVisibility.PRIVATE.name)
+                    dataset = self.generate_dataset(
+                        self.session,
+                        collection=collection,
+                        processing_status=processing_status,
+                        curator_tag=curator_tag,
+                    )
+                    test_url = f"/curation/v1/collections/{collection.id}/datasets/"
+                    if query == "dataset_uuid":
+                        query_string = {query: dataset.id}
+                    elif query == "curator_tag":
+                        query_string = {query: curator_tag}
+                    headers = auth() if callable(auth) else auth
+                    response = self.app.delete(test_url, headers=headers, query_string=query_string)
```

Comment:
Perfect, I was going to suggest this since it is in accordance with [my post](https://czi-sci.slack.com/archives/C02UGUV5R3J/p1652995597685589).

Changes need to be made to the spec, though

Reply:
Anyways I think we would be best off continuing the theme (see [my post](https://czi-sci.slack.com/archives/C02UGUV5R3J/p1652995597685589)) of removing from the curator's workflow the concern of deciding whether they are using the tag or the uuid: instead of having the `curator_tag` and `dataset_uuid` url query parameter options, I think there should be a single url parameter here, perhaps called `identifier`, which the curator could submit a tag or a dataset uuid to.

Reply:
[query_string](https://werkzeug.palletsprojects.com/en/2.1.x/test/) is there. It's just burried.

Reply:
> Anyways I think we would be best off continuing the theme (see [my post](https://czi-sci.slack.com/archives/C02UGUV5R3J/p1652995597685589)) of removing from the curator's workflow the concern of deciding whether they are using the tag or the uuid: instead of having the `curator_tag` and `dataset_uuid` url query parameter options, I think there should be a single url parameter here, perhaps called `identifier`, which the curator could submit a tag or a dataset uuid to.

We can save that for V2 if the curators complain.